bio-primer-removal/main.py

```diff
@@ -0,0 +1,114 @@
+from typing import Tuple, Optional
+from pathlib import Path
+from zipfile import ZipFile, ZIP_DEFLATED
+
+import logging
+
+from coretex import Experiment, CustomDataset, CustomSample, folder_manager, SequenceDataset, SequenceSample
+from coretex.project import initializeProject
+from coretex.bioinformatics import cutadaptTrim, isPairedEnd
+
+
+def forwardMetadata(sample: CustomSample, outputDataset: SequenceDataset) -> None:
+    sample.unzip()
+
+    metadataZip = folder_manager.temp / "_metadata.zip"
+    with ZipFile(metadataZip, 'w', ZIP_DEFLATED) as archive:
+        for path in sample.path.iterdir():
+            archive.write(path, path.name)
+
+    if CustomSample.createCustomSample("_metadata", outputDataset.id, metadataZip) is None:
```

Comment:
why is it needed to zip the metadata sample here? why can't we use `sample.zipPath` and upload that?